### DISCUSSION TITLE
Quickpassthrough 2.0 is slowly on its way

### DISCUSSION THREAD

**Discussion by HikariKnight** [HikariKnight/quickpassthrough#17] [Announcements]
Hi all, some of you might have spotted the [dev branch](https://github.com/HikariKnight/quickpassthrough/tree/dev) that has been lurking in the git repo while the main branch has not had any updates in a very long time.

Well back in April i made a small project aptly named [ls-iommu](https://github.com/HikariKnight/ls-iommu) that is aimed to replace the old ls-iommu.sh script we have used for many years, you know... this one
```bash
#!/bin/bash
for d in /sys/kernel/iommu_groups/*/devices/*; do
  n=${d#*/iommu_groups/*}; n=${n%%/*}
  printf 'IOMMU Group %s ' "$n"
  lspci -nns "${d##*/}"
done
```

The ls-iommu script is nice and small but I had many gripes with it:
- The output cannot be sorted, making it messy
- To get specific info from it you will have to use a fun chain of grep, sed and/or perl, etc
- It was not very user friendly for more detailed info
- Without modifications it would not show kernel drivers in use

I then decided to write a version in go with some help from @digitalsparky that would make getting the info easier, enter [ls-iommu](https://github.com/HikariKnight/ls-iommu)
So what does this have to do with Quickpassthrough 2.0? Well ls-iommu is an integral part to quickpassthrough and I always wanted to make a proper TUI for the project and since i had a "minimal dependency" philosophy for the project, go would be perfect for that when using `CGO_ENABLED=0`.

The bash version of quickpassthrough (now Quickpassthrough 1) was a nice proof of concept that showed it is possible to simplify gpu passthrough, it has done its job and now it is time to make the actual project in something more concrete.

However due to a bunch of things going on in my life, I got only the core functionality ported to go earlier this year. The project still had no fully functional polished TUI, even though it had a rudimentary one spliced together using [BubbleTea](https://github.com/charmbracelet/bubbletea), which is a really nice framework but way overkill for what this project needs (unless someone skilled in it can mock up something nice that makes sense!) so this month i have been working on just keeping the logger from BubbleTea but using [gocliselect](https://github.com/nexidian/gocliselect) for the actual interactive part of the TUI as all it needs is just something akin to a ChoiceInput from Qt and some "YesNo" dialogs, this is very easy to do with gocliselect without adding complexity.

All the things that have kept me occupied from working on this has now been dealt with, meaning once my real life work calms down (it is winding down already hence the recent commits to the dev branch), I will be able to hopefully get Quickpassthrough 2.0 released either this year or early next year :eyes:

If anyone is interested in helping and contributing, feel free to do so, any help even minor is appreciated!

**Comment by digitalsparky** [comment]
May I also suggest adding UPX to the build steps on github actions - this shrinks the binary significantly (as well as using the -s -w flags IIRC).

I do it through travis-ci at the  moment on a few things, but you can see the build steps here as an example:

https://github.com/digitalsparky/geowall/blob/master/.travis.yml

**Reply by HikariKnight** [reply]
there shouldnt be any automated builds yet anyway until everything is done, plan was to just use everything similar to how we did it in ls-iommu

**Reply by digitalsparky** [reply]
of course, as you can see in the example:

on:
    repo: digitalsparky/geowall
    tags: true

So it will only build and push this job on pushed tags/releases. :).

**Reply by HikariKnight** [reply]
will look into it when we get closer for a first release :smile:
